### DISCUSSION TITLE
Feature quest: Support regexp in EXCLUDE clause

### DISCUSSION THREAD

**Discussion by jaceksan** [duckdb/duckdb#10666] [Ideas]
Especially valuable when it is not possible to exclude columns from being loaded into DB and then there is no way how to skip them when executing SELECT from the table.

Example:
It is not possible to suppress loading of `fields__customfield*` columns when using Meltano `tap-jira` plugin.
Being able to execute `SELECT * EXCLUDE('fields__customfield.*')` would be awesome!

**Comment by jaceksan** [comment]
Btw. the issue is also valid for skipping "technical" columns produced by tools like Meltano or dbt, e.g. `sdc_*` or `dbt_*`.

**Reply by Tishj** [reply]
I think you're looking for <https://duckdb.org/docs/sql/expressions/star#columns-lambda-function>

**Reply by jaceksan** [reply]
You made my day!
This works:
```
SELECT COLUMNS(c -> c NOT LIKE 'fields__custom%') FROM jira_input_stage.issues limit 10;
```

Thank you!

**Comment by EED85** [comment]
Short Question: can this be closed? Makes searching for open discussions more diffcult

**Comment by jaceksan** [comment]
Feel free to close it, the alternative solution with LAMBDA is perfect.

**Reply by soerenwolfers** [reply]
he was meaning that _you_ should close it. Only the original poster or repository maintainers can close discussions.
